### PR TITLE
Update REST API plugin to 1.2.1

### PR DESCRIPTION
**Pull request by Redor**
Fixed an cluster issue by create a chat room

Sry for the whole diff, something went wrong with my IDE

### CODE REVIEW

**Comment by akrherz** [comment]
Is the minVersion of 3.10.0 you have set correct?

**Comment by Redor** [comment]
Yes, that is correct.

**Comment by tevans** [comment]
Hmm - hard to tell what was changed here. Is it possible to clean up the diff?

File: src/plugins/restAPI/src/java/org/jivesoftware/openfire/plugin/rest/controller/MUCRoomController.java [hunk 1]
```diff
@@ -1,547 +1,556 @@
...
+		room.setCanOccupantsInvite(mucRoomEntity.isCanOccupantsInvite());
+		room.setChangeNickname(mucRoomEntity.isCanChangeNickname());
+		room.setModificationDate(mucRoomEntity.getModificationDate());
+		room.setLogEnabled(mucRoomEntity.isLogEnabled());
+		room.setLoginRestrictedToNickname(mucRoomEntity.isLoginRestrictedToNickname());
+		room.setMaxUsers(mucRoomEntity.getMaxUsers());
+		room.setMembersOnly(mucRoomEntity.isMembersOnly());
+		room.setModerated(mucRoomEntity.isModerated());
+		
+		// Fire RoomUpdateEvent if cluster is started
```
Comment (Redor) [inline src/plugins/restAPI/src/java/org/jivesoftware/openfire/plugin/rest/controller/MUCRoomController.java]: Follow lines are added.

File: src/plugins/restAPI/src/java/org/jivesoftware/openfire/plugin/rest/controller/MUCRoomController.java [hunk 2]
```diff
@@ -1,547 +1,556 @@
...
+		if (ClusterManager.isClusteringStarted()) {
```
Comment (Redor) [inline src/plugins/restAPI/src/java/org/jivesoftware/openfire/plugin/rest/controller/MUCRoomController.java]: Sry I could not manage that to clean up the diff :/
Follow line are added

**Comment by Redor** [comment]
Sry I could not manage that to clean up the diff :/
Follow line are added by creating a new chat Channel:

```
// Fire RoomUpdateEvent if cluster is started
if (ClusterManager.isClusteringStarted()) {
  CacheFactory.doClusterTask(new RoomUpdatedEvent((LocalMUCRoom) room));
}
```

**Comment by tevans** [comment]
OK by me, +1

**Comment by VeronicaSL** [comment]
I have the same problem: https://community.igniterealtime.org/thread/57607

Any solutions?, I have installed Openfire 3.10.3

**Comment by Geocomt** [comment]
I make clean install to Openfire 3.10.3 and before add this plugin (ApiRest - 2.1.1).
In my plugin list show correct, but in Server->Server Settings" in the admin control website, show a blank page.

**Comment by Redor** [comment]
It seems to be a compatibility problem in the REST API v. 1.2.1 . Please use the latest working version 1.1.7  http://bamboo.igniterealtime.org/artifact/OPENFIRE-NIGHTLYPLUGINS/shared/build-1 817/Project-Plugins/restAPI.jar

**Comment by VeronicaSL** [comment]
Thanks IT WORKS.

But the link is broken, the correct link is:
http://bamboo.igniterealtime.org/artifact/OPENFIRE-NIGHTLYPLUGINS/shared/build-01817/Project-Plugins/restAPI.jar

**Comment by Rajeshwar90** [comment]
The above link also does not work.Can anybody please provided the correct link for the rest api jar that is compatible with the Openfire version 3.10.2. The latest RestAPI  jar is not compatible

**Comment by akrherz** [comment]
@Rajeshwar90  try this: http://bamboo.igniterealtime.org/browse/OPENFIRE-PB3-23/artifact/shared/Project-Plugins/

**Comment by Rajeshwar90** [comment]
Thanks akrherz.I have installed RestAPI. Now when I hit the openfire server for users  using the secret key i am getting 401 unauthorized. Can you please guide me.

http://openfire.smartchk.net:9090/plugins/restapi/v1/users
Headers:
Authorization: "secretkey"
Accept: application/json

**Comment by akrherz** [comment]
@Rajeshwar90 This pull request is not a support forum for your problems.  Please use the Community Forums.  https://community.igniterealtime.org

**Comment by Rajeshwar90** [comment]
Ok Akrherz.I will use the community forum.

**Pull request merged**